technical/trendline.py

```diff
@@ -5,127 +5,86 @@
 """
 
 
-def gentrends(dataframe, field="close", window=1 / 3.0, charts=False):
-    """
-    Returns a Pandas dataframe with support and resistance lines.
-
-    :param dataframe: incomming data matrix
-    :param field: for which column would you like to generate the trendline
-    :param window: How long the trendlines should be. If window < 1, then it
-                   will be taken as a percentage of the size of the data
-    :param charts: Boolean value saying whether to print chart to screen
-    """
-
-    x = dataframe[field]
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
 
-    import numpy as np
-    import pandas as pd
-
-    x = np.array(x)
+def gentrends(dataframe, field="close", window=1 / 3.0, charts=False):
+    x = np.array(dataframe[field])
 
     if window < 1:
         window = int(window * len(x))
 
-    max1 = np.where(x == max(x))[0][0]  # find the index of the abs max
-    min1 = np.where(x == min(x))[0][0]  # find the index of the abs min
+    max1_idx = np.argmax(x)
+    min1_idx = np.argmin(x)
 
-    # First the max
-    if max1 + window >= len(x):
-        max2 = max(x[0 : (max1 - window)])
+    max1 = x[max1_idx]
+    min1 = x[min1_idx]
+
+    if max1_idx + window >= len(x):
+        max2 = np.max(x[0 : (max1_idx - window)])
     else:
-        max2 = max(x[(max1 + window) :])
+        max2 = np.max(x[(max1_idx + window) :])
 
-    # Now the min
-    if min1 - window <= 0:
-        min2 = min(x[(min1 + window) :])
+    if min1_idx - window <= 0:
+        min2 = np.min(x[(min1_idx + window) :])
     else:
-        min2 = min(x[0 : (min1 - window)])
-
-    # Now find the indices of the secondary extrema
-    max2 = np.where(x == max2)[0][0]  # find the index of the 2nd max
-    min2 = np.where(x == min2)[0][0]  # find the index of the 2nd min
-
-    # Create & extend the lines
-    maxslope = (x[max1] - x[max2]) / (max1 - max2)  # slope between max points
-    minslope = (x[min1] - x[min2]) / (min1 - min2)  # slope between min points
-    a_max = x[max1] - (maxslope * max1)  # y-intercept for max trendline
-    a_min = x[min1] - (minslope * min1)  # y-intercept for min trendline
-    b_max = x[max1] + (maxslope * (len(x) - max1))  # extend to last data pt
-    b_min = x[min1] + (minslope * (len(x) - min1))  # extend to last data point
-    maxline = np.linspace(a_max, b_max, len(x))  # Y values between max's
-    minline = np.linspace(a_min, b_min, len(x))  # Y values between min's
-
-    # OUTPUT
-    trends = np.transpose(np.array((x, maxline, minline)))
+        min2 = np.min(x[0 : (min1_idx - window)])
+
+    max2_idx = np.where(x == max2)[0][0]
+    min2_idx = np.where(x == min2)[0][0]
+
+    maxslope = (max1 - max2) / (max1_idx - max2_idx)
+    minslope = (min1 - min2) / (min1_idx - min2_idx)
+    a_max = max1 - (maxslope * max1_idx)
+    a_min = min1 - (minslope * min1_idx)
+    b_max = max1 + (maxslope * (len(x) - max1_idx))
```

Comment:
I don't see why we would remove the comments (not just on the highlighted lines).
that seems to make following (and understanding) this code quite more complicated (it's not immediately clear what `a_max` is supposed to contain - for example), especially for someone not 100% familiar with how this code is supposed to work.


Please revert / keep the comments.

Reply:
I will readd the comments in order to make it beginner friendly